src/polycap-out.c

```diff
@@ -94,15 +95,24 @@ void polycap_transmission_efficiencies_write_hdf5(const char *filename, polycap_
 	}
 	
 	//Write simulated polycap exit coordinates
+	//Copy coordiante data to temporary array for straightforward HDF5 writing
+	data_temp[0] = malloc(sizeof(double)*efficiencies->images->i_exit);
+	data_temp[1] = malloc(sizeof(double)*efficiencies->images->i_exit);
+	for(j=0;j<efficiencies->images->i_exit;j++){
+		data_temp[0][j] = efficiencies->images->pc_exit_coords[0][j];
+		data_temp[1][j] = efficiencies->images->pc_exit_coords[1][j];
+printf("*x: %f, y:%f\n",data_temp[0][j],efficiencies->images->pc_exit_coords[1][j]);
```

Comment:
Laat mij raden: deze `printf` produceert goede waarden, maar ze worden verkeerd naar de hdf5 file geschreven?

HDF5 verwacht dat je continue blocks memory naar de file schrijft, maar `data_temp` is dat niet. `data_temp[0]` is op zich continue, en `data_temp[1]` ook, maar ze zullen tezamen geen continu blok vormen aangezien ze gealloceerd zijn met twee aparte malloc's en dus niet opeenvolgend in het geheugen zitten (tenzij ge onwaarschijnlijk veel geluk hebt, maar daar zou ik echt niet op rekenen 😄 )

Reply:
Sorry was al aan het schrijven voor ik uw comment zag.

Wat je moet doen is iets als:

```c
double *data_temp = malloc(sizeof(double) * efficiencies->images->i_exit * 2);
```

Vervolgens omdat C [row-major ordering](https://en.wikipedia.org/wiki/Row-_and_column-major_order) gebruikt moet je dan je lus schrijven als:

```c
for(j=0;j<efficiencies->images->i_exit;j++){
 	data_temp[j] = efficiencies->images->pc_exit_coords[0][j];
 	data_temp[j + efficiencies->images->i_exit] = efficiencies->images->pc_exit_coords[1][j];
}
```

Reply:
idd. Dat lijkt me het probleem.
Maar hoe maak ik een data_temp array die wel in een continue block schrijft? :( 
Op zich zou ik iets proberen als `double data_temp[2][sum_istart]` maar kvrees dat dat mis loopt daar bij compilatie op zich sum_istart niet gekend is. Of is dat geen probleem?

Reply:
Mja, dat is mogelijk in [C99](c99 variable length array) maar dat is nogal aan de gevaarlijke kant want als je een te grote array probeert te definiëren op die manier kan je een stack overflow krijgen. Met malloc krijg je tenminste een NULL als er niet genoeg geheugen beschikbaar is.

En je zou sneller in de problemen raken dan je denkt want de stack is sterk gelimiteerd tov de heap (malloc).

Reply:
Hm, oke :) Had al eens iets gelijkaardigs geprobeerd maar kzat fout, wss door de row-major ordering en wat geklungel in meerdere for loops. Jouw manier lijkt eleganter :) Eens zien of het werkt.